spec/services/candidate_interface/decoupled_references/request_reference_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+RSpec.describe CandidateInterface::DecoupledReferences::RequestReference do
+  describe '#call' do
+    let(:reference) { create(:reference, feedback_status: 'not_requested_yet') }
+    let(:execute_service) { described_class.new.call(reference, flash) }
+    let(:mail) { instance_double(ActionMailer::MessageDelivery, deliver_later: true) }
+    let(:flash) { ActionDispatch::Flash::FlashHash.new }
```

Comment:
Depending on flash feels a bit weird to me — does it feel weird to you?

Reply:
Yea it definitely does. I've not come across a flash being passed into the service. What approach would you have taken here? Should I be injecting a double in?

Reply:
I think this service shouldn't depend on flash at all — only the controller should know about the flash, and it should use the return value of this thing to make decisions on what to do with the flash. Really hardly a problem here because it's such a tiny object! But in general it's good to avoid this kind of coupling from controllers to the service layer